safekeeper/src/wal_storage.rs

```diff
@@ -379,6 +379,12 @@ impl Storage for PhysicalStorage {
             );
         }
 
+        // Quick exit if nothing to do to avoid writing up to 16 MiB of zeros on
+        // disk (this happens on each connect).
+        if end_pos == self.write_lsn {
+            return Ok(());
```

Comment:
We probably should also flush the file here (or check that `write_lsn == flush_lsn` and skip the flush)

Reply:
It won't harm, but why? This functions just truncates, we don't expect here any guarantees about flushed part AFAIS.

Reply:
AFAIU `truncate_lsn` should have flush guarantees, otherwise we can (1. start streaming ahead of `flush_lsn` and lose some WAL) or (2. "untruncate" WAL). But I'm not sure about this, so I'll think a bit more if anything bad can potentially happen.